engine/gamesys/src/gamesys/components/comp_collision_object.cpp

```diff
@@ -1006,6 +1001,7 @@ namespace dmGameSystem
             physics_socket = dmPhysics::GetSocket2D(physics_context->m_Context2D);
 
         }
+
```

Comment:
One of the root causes of this issue, is that we only have one `@physics` socket, and each `CollisionWorld*` is flushing the message queue. It means we don't have a good way to provide better context  for handling the messages.

See the comment above this function for previous suggested solutions (e.g. one socket per physics world)